### PR TITLE
UR ams ros2_control support for rox platform

### PR DESCRIPTION
**Pull request by AdarshKaran**
Added ros2_control to urdf for arm support, modified launch files to support this and tested with neo_rox_moveit2 package

### CODE REVIEW

File: rox_bringup/launch/bringup_sim_launch.py [hunk 1]
```diff
@@ -46,7 +51,7 @@ def execution_stage(context: LaunchContext, frame_type, rox_type, arm_type, use_
         PythonLaunchDescriptionSource(
             os.path.join(get_package_share_directory('ros_gz_sim'), 'launch', 'gz_sim.launch.py')
         )
-        , launch_arguments={'ign_args': ['-r ', default_world_path]}.items()
+        , launch_arguments={'gz_args': ['-r ', default_world_path]}.items()
```
Comment (padhupradheep) [inline rox_bringup/launch/bringup_sim_launch.py:54]: could you also change all the other ign references to gz ?

**Review by padhupradheep — COMMENTED** [review]
Overall looks good, ~did you check it with MoveIt?~

Just a small change is required.

File: rox_bringup/launch/bringup_sim_launch.py [hunk 2]
```diff
@@ -86,8 +91,29 @@ def execution_stage(context: LaunchContext, frame_type, rox_type, arm_type, use_
...
+    )
+
+    launch_actions.append(start_robot_state_publisher_cmd)
+    if arm_type != '':
+        launch_actions.append(joint_state_broadcaster_spawner)
+        launch_actions.append(initial_joint_controller_spawner_stopped)
+    launch_actions.append(ignition)
+    launch_actions.append(gz_bridge)
+    launch_actions.append(teleop)
+    launch_actions.append(spawn_robot)
```
Comment (padhupradheep) [inline rox_bringup/launch/bringup_sim_launch.py:114]: Only needed to be added to launch actions when ur arm is selected.

Comment (AdarshKaran) [reply rox_bringup/launch/bringup_sim_launch.py:114]: launch_actions.append(start_robot_state_publisher_cmd)
    if arm_type != '':
        launch_actions.append(joint_state_broadcaster_spawner)
        launch_actions.append(initial_joint_controller_spawner_stopped)
        launch_actions.append(gz)
        launch_actions.append(gz_bridge)
        launch_actions.append(teleop)
        launch_actions.append(spawn_robot)
        
        like this anna? so without arm only robot state pub shoudl be run?

File: rox_bringup/launch/bringup_sim_launch.py (hunk 1 above)
Comment (AdarshKaran) [reply rox_bringup/launch/bringup_sim_launch.py:54]: this seems to be the only one anna

File: rox_bringup/launch/bringup_sim_launch.py (hunk 2 above)
Comment (padhupradheep) [reply rox_bringup/launch/bringup_sim_launch.py:114]: Yes, something like this. People should be just able to use ROX standalone without arm on the top

Comment (AdarshKaran) [reply rox_bringup/launch/bringup_sim_launch.py:114]: I have fixed it Pradheep, there was a typo in that condition. Now it should work fine

**Comment by padhupradheep** [comment]
Did you notice when the robot navigates, the robot tips over?

**Comment by AdarshKaran** [comment]
i have fixed the issue now by updating the mass in the frame.xacro file Pradheep and pushed the changes

**Pull request merged**